core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -0,0 +1,936 @@
+        // Check for throttled response
+        ShareFetchResponse resp7 = context7.throttleResponse(100);
+        assertEquals(Errors.NONE, resp7.error());
+        assertEquals(100, resp7.throttleTimeMs());
+
+        assertErroneousAndValidTopicIdPartitions(context7.getErroneousAndValidTopicIdPartitions(), Arrays.asList(tpNull1, tpNull2), Arrays.asList(tp0, tp1));
+
+        // Close the subsequent share session.
+        ShareFetchContext context8 = sharePartitionManager.newContext(groupId, Collections.emptyMap(), EMPTY_PART_LIST,
+                new ShareFetchMetadata(reqMetadata2.memberId(), ShareFetchMetadata.FINAL_EPOCH));
+        assertEquals(FinalContext.class, context8.getClass());
+        assertEquals(0, cache.size());
+
+
+        assertErroneousAndValidTopicIdPartitions(context8.getErroneousAndValidTopicIdPartitions(), Collections.emptyList(), Collections.emptyList());
+        // Check for throttled response
+        ShareFetchResponse resp8 = context8.throttleResponse(100);
+        assertEquals(Errors.NONE, resp8.error());
+        assertEquals(100, resp8.throttleTimeMs());
+    }
+
+    @Test
+    public void testShareFetchContextResponseSize() {
+        Time time = new MockTime();
+        ShareSessionCache cache = new ShareSessionCache(10, 1000);
+        SharePartitionManager sharePartitionManager = SharePartitionManagerBuilder.builder()
+                .withCache(cache).withTime(time).build();
+        Map<Uuid, String> topicNames = new HashMap<>();
+        Uuid tpId0 = Uuid.randomUuid();
+        Uuid tpId1 = Uuid.randomUuid();
+        topicNames.put(tpId0, "foo");
+        topicNames.put(tpId1, "bar");
+        TopicIdPartition tp0 = new TopicIdPartition(tpId0, new TopicPartition("foo", 0));
+        TopicIdPartition tp1 = new TopicIdPartition(tpId0, new TopicPartition("foo", 1));
+        TopicIdPartition tp2 = new TopicIdPartition(tpId1, new TopicPartition("bar", 0));
+        TopicIdPartition tp3 = new TopicIdPartition(tpId1, new TopicPartition("bar", 1));
+
+        String groupId = "grp";
+
+        // Create a new share session with an initial share fetch request
+        Map<TopicIdPartition, ShareFetchRequest.SharePartitionData> reqData2 = new LinkedHashMap<>();
+        reqData2.put(tp0, new ShareFetchRequest.SharePartitionData(tp0.topicId(), 100));
+        reqData2.put(tp1, new ShareFetchRequest.SharePartitionData(tp1.topicId(), 100));
+
+        // For response size expected value calculation
+        ObjectSerializationCache objectSerializationCache = new ObjectSerializationCache();
+        short version = ApiKeys.SHARE_FETCH.latestVersion();
+
+        ShareFetchMetadata reqMetadata2 = new ShareFetchMetadata(Uuid.randomUuid(), ShareFetchMetadata.INITIAL_EPOCH);
+        ShareFetchContext context2 = sharePartitionManager.newContext(groupId, reqData2, EMPTY_PART_LIST, reqMetadata2);
+        assertEquals(context2.getClass(), ShareSessionContext.class);
```

Comment:
Can you please check in other tests as well where we have this discrepancy.

```suggestion
        assertEquals(ShareSessionContext.class, context2.getClass());
```